icelake/src/io_v2/file_writer/base_file_writer.rs

```diff
@@ -276,20 +272,40 @@ mod prometheus {
         /// Write a record batch. The `DataFileWriter` will create a new file when the current row num is greater than `target_file_row_num`.
         async fn write(&mut self, batch: &RecordBatch) -> Result<()> {
             self.inner.write(batch).await?;
-            let last_metrics =
-                std::mem::replace(&mut self.cur_metrics, self.inner.metrics().clone());
+            let last_metrics = std::mem::replace(&mut self.cur_metrics, self.inner.metrics());
             {
                 let delta =
                     (self.cur_metrics.unflush_data_file - last_metrics.unflush_data_file) as i64;
                 assert!(delta >= 0);
-                self.unflush_data_size.add(delta as u64);
+                self.unflush_data_file.add(delta as u64);
             }
             Ok(())
         }
 
         /// Complte the write and return the list of `DataFile` as result.
         async fn close(self) -> Result<Vec<Self::R>> {
-            self.inner.close().await
+            let res = self.inner.close().await?;
+            let delta = (res.len() - self.cur_metrics.unflush_data_file) as i64;
+            assert!(delta >= 0);
+            self.unflush_data_file.add(delta as u64);
+            Ok(res)
+        }
+    }
+
+    impl<B: FileWriterBuilder> SingleFileWriterStatus for BaseFileWriterWithMetrics<B>
```

Comment:
This seems interesting. If rolling file writer also needs this metrics, how about make it part of `FileWriter`?

Reply:
Yes, make it part of FileWriter is more simple. But we still need it in IcebergWriter. How about 
```
#[async_trait::async_trait]
pub trait FileWriter: Send + 'static + SingleFileWriterStatus {
    type R: FileWriteResult;
    async fn write(&mut self, batch: &RecordBatch) -> Result<()>;
    async fn close(self) -> Result<Vec<Self::R>>;
}
```
The benefit of it is that we don't need to maintain the duplicate interface: one in FileWriter and one for the SingleFileWriterStatus trait.🤔 And we can guarantee that all the FileWriter implement this interface.

Reply:
LGTM. How about renaming it to `CurrentFileStatus`? It sounds more clear.